### PR TITLE
RBN-89: Fetching photograph from db

### PR DESCRIPTION
**Pull request by Aleksandra-Romanic**
### What did you implement?

Implemented an endpoint for retrieving book cover image by the book id. Updated BookController and BookService to handle retrieving logic.
Added unit tests for both the controller and service methods, and integration test to verify the endpoint's behavior.

### How can we verify the changes?

Changes can be verified by running the application locally, testing the endpoint in swagger and checking the tests.

### More info

- Jira: [RBC-89](https://productdock.atlassian.net/jira/software/projects/RBN/boards/11?selectedIssue=RBN-89)

### CODE REVIEW

File: rbc/mongo-init.js [hunk 1]
```diff
@@ -13,7 +12,7 @@ db.books.insertMany([
         "fullName": "Kristina Chodorow"
       }
     ],
-    "imageUrl": "image",
+    "imageUrl": "/Users/pd/Documents/images/BookCover1.svg",
```
Comment (tamastarjan) [inline rbc/mongo-init.js:15]: Do these images have to exist before setting up the local development environment? If this is the case, then please update the README.md with this information.

File: rbc/src/main/java/com/library/rbc/controller/BookController.java [hunk 2]
```diff
@@ -48,4 +49,12 @@ public Page<BookDto> returnBooksBy(@ParameterObject @PageableDefault(size = 12)
       @RequestParam(required = false) List<String> bookStatuses) {
     return bookService.getBooksBy(pageable, bookCategories, bookStatuses);
   }
+
+  @GetMapping("/photo/{bookId}")
```
Comment (tamastarjan) [inline rbc/src/main/java/com/library/rbc/controller/BookController.java]: Endpoint suggestion: `@GetMapping("/{bookId}/image")`

File: rbc/src/main/java/com/library/rbc/service/BookService.java [hunk 3]
```diff
@@ -88,6 +95,21 @@ private List<BookStatusDto> convertStringsToBookStatusesDto(List<String> bookSta
         }).toList();
   }
 
-
+  public ImageWithMediaTypeDto getBookImageById(String id) {
+    BookDto bookDto = bookMapper.bookToBookDto(
+        bookRepository.findById(id).orElseThrow(
+            () -> new BookNotFoundException("Book with ID " + id + " was not found.")));
```
Comment (tamastarjan) [inline rbc/src/main/java/com/library/rbc/service/BookService.java]: Is this mapping needed? Can we get the image url from the book entity?

File: rbc/src/test/java/com/library/rbc/controller/bookcontroller/BookControllerShould.java [hunk 4]
```diff
@@ -93,4 +97,19 @@ void returnBooksBy(){
...
+  void testReturnPhotoBy() {
+    byte[] imageBytes = new byte[]{};
+    ImageWithMediaTypeDto mockImageDto = new ImageWithMediaTypeDto(
+        imageBytes, MediaType.IMAGE_JPEG);
+
+    when(bookService.getBookImageById(BOOK_ID)).thenReturn(mockImageDto);
+    ResponseEntity<byte[]> response = bookController.returnPhotoBy(BOOK_ID);
+
+    assertArrayEquals(imageBytes, response.getBody());
+
```
Comment (tamastarjan) [inline rbc/src/test/java/com/library/rbc/controller/bookcontroller/BookControllerShould.java]: empty line

**Review by tamastarjan — APPROVED** [review]
Review: LGTM

**Review by teodoraBr — APPROVED** [review]

**Pull request merged**